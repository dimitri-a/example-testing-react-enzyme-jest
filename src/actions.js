export const handleStartCar = () => {
  console.log('car has started');
};
