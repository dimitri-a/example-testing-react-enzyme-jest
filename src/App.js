import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Car from './Car';
import * as actions from './actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class App extends Component {
  render() {
    const { handleStartCar, isRunning } = this.props;
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          {isRunning ? 'The car is running  ' : 'Click to start the car'}
        </p>
        <Car color='blue' handleStartCar={handleStartCar} />
      </div>
    );
  }
}
App.propTypes = {
  // Redux store
  isRunning: PropTypes.bool,
  // advertiseActions
  handleStartCar: PropTypes.func
};
App.defaultProps = {
  isRunning: false,
  handleStartCar: () => {
    console.log('car started');
  }
}
export default connect(
  (state) => ({
    isRunning: state.isRunning
  }),
  {...actions}
)(App);
