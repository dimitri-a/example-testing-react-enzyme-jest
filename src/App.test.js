import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import Car from './Car';
import * as actions from './actions';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  shallow(<App />);
});
it('renders welcome message', () => {
  const wrapper = shallow(<App />);
  const welcome = <h2>Welcome to React</h2>;
  // expect(wrapper.contains(welcome)).to.equal(true);
  expect(wrapper.contains(welcome)).toEqual(true);
});
it('renders Car', () => {
  const props = {
    isRunning: false,
    handleStartCar: () => {}
  };
  const wrapper = shallow(<App {...props} />);
  const car = wrapper.find(Car);
  expect(car).toHaveLength(1);
  expect(car.prop('handleStartCar')).toEqual(props.handleStartCar);
});
it('matches snapshot', () => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
