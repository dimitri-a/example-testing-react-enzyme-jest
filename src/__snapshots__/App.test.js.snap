exports[`<App /> matches snapshot 1`] = `
<App
  handleStartCar={[Function]}
  isRunning={false}>
  <div
    className="App">
    <div
      className="App-header">
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg" />
      <h2>
        Welcome to React
      </h2>
    </div>
    <p
      className="App-intro">
      Click to start the car
    </p>
    <Car
      color="blue"
      handleStartCar={[Function]}>
      <div>
        <h1
          id="car--heading"
          style={
            Object {
              "color": "blue",
            }
          }>
          I am a 
          blue
           car
        </h1>
        <Engine
          type="diesel">
          <div>
            <span
              id="engine--type">
              diesel
               type
            </span>
          </div>
        </Engine>
        <button
          id="car--start"
          onClick={[Function]}
          type="button">
          Start
        </button>
      </div>
    </Car>
  </div>
</App>
`;

exports[`<App /> matches snapshot 2`] = `
<App
  handleStartCar={[Function]}
  isRunning={true}>
  <div
    className="App">
    <div
      className="App-header">
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg" />
      <h2>
        Welcome to React
      </h2>
    </div>
    <p
      className="App-intro">
      The car is running  
    </p>
    <Car
      color="blue"
      handleStartCar={[Function]}>
      <div>
        <h1
          id="car--heading"
          style={
            Object {
              "color": "blue",
            }
          }>
          I am a 
          blue
           car
        </h1>
        <Engine
          type="diesel">
          <div>
            <span
              id="engine--type">
              diesel
               type
            </span>
          </div>
        </Engine>
        <button
          id="car--start"
          onClick={[Function]}
          type="button">
          Start
        </button>
      </div>
    </Car>
  </div>
</App>
`;
